s4e-backend/src/test/java/pl/cyfronet/s4e/controller/GroupControllerTest.java

```diff
@@ -0,0 +1,202 @@
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.post;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.put;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status;
+import static pl.cyfronet.s4e.Constants.API_PREFIX_V1;
+
+@AutoConfigureMockMvc
+@BasicTest
+@Slf4j
+public class GroupControllerTest {
+
+    @Autowired
+    private InstitutionRepository institutionRepository;
+
+    @Autowired
+    private GroupRepository groupRepository;
+
+    @Autowired
+    private AppUserRepository appUserRepository;
+
+    @Autowired
+    private GroupService groupService;
+
+    @Autowired
+    private ObjectMapper objectMapper;
+
+    @Autowired
+    private SlugService slugService;
+
+    private String slugInstitution = "";
+
+    @Autowired
+    private MockMvc mockMvc;
+
+    @BeforeEach
+    public void beforeEach() {
+        institutionRepository.deleteAll();
+        appUserRepository.deleteAll();
+
+        String test_institution = "Test Institution";
+        slugInstitution = slugService.slugify(test_institution);
+        institutionRepository.save(Institution.builder()
+                .name(test_institution)
+                .slug(slugInstitution)
+                .build());
+    }
+
+    @Test
+    public void shouldCreateGroupWithoutMembers() throws Exception {
+        CreateGroupRequest groupRequest = CreateGroupRequest.builder()
+                .name("CreateGroupTest")
+                .build();
+
+        mockMvc.perform(post(API_PREFIX_V1+"/institutions/{institution}/groups", slugInstitution)
+                .contentType(MediaType.APPLICATION_JSON)
+                .content(objectMapper.writeValueAsBytes(groupRequest)))
+                .andExpect(status().isOk());
+
+        assertThat(groupRepository.findBySlugAndInstitution_Slug("creategrouptest",slugInstitution).isPresent(), is(true));
+    }
+
+    @Test
+    public void shouldCreateGroupWithMembers() throws Exception {
+        String email = "some@email.pl";
+        appUserRepository.save(AppUser.builder()
+                .name("Name")
+                .surname("Surname")
+                .email(email)
+                .password("admin123")
+                .enabled(true)
+                .build());
+        Set<String> members = new HashSet<>();
+        members.add(email);
+        CreateGroupRequest groupRequest = CreateGroupRequest.builder()
+                .name("CreateGroupTest")
+                .membersEmails(members)
+                .build();
+
+        mockMvc.perform(post(API_PREFIX_V1+"/institutions/{institution}/groups", slugInstitution)
+                .contentType(MediaType.APPLICATION_JSON)
+                .content(objectMapper.writeValueAsBytes(groupRequest)))
+                .andExpect(status().isOk());
+
+        assertThat(groupRepository.findBySlugAndInstitution_Slug(
+                "creategrouptest",slugInstitution).isPresent(), is(true));
```

Comment:
I've found a hamcrest extension, which adds support for optionals: https://github.com/npathai/hamcrest-optional. Might be worth a shot?

Reply:
#222